### ISSUE TITLE
Add Git integration as separate SLR library feature

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#12630]
The library should have a property "Git"

- auto pull, auto commit, and auto push

Currently, git is availble in the "File" menu. There should be an automatic setting - in the library preferences. -- No need to set the git remote, that one is set in `.git` config (on the git side)

~~Currently, this is a feature of the SLR thing. It should be factored out.~~

~~`org.jabref.logic.git.GitHandler`~~

**Comment by koppor** [comment]
Before working on this issue, double check the existing code - much has been going on since the last 3 years.

**Comment by D-Prasanth-Kumar** [comment]
> Before working on this issue, double check the existing code - much has been going on since the last 3 years.

yeah after checking and seeing what is going on , i tried to assign to me. so the issue is like there is no specific git section in library properties , and we need to add the git section in it and when we select it and save the files in that library commit and push should be done automatically. is it something like it right.

**Comment by koppor** [comment]
> yeah after checking and seeing what is going on , i tried to assign to me. so the issue is like there is no specific git section in library properties , and we need to add the git section in it and when we select it and save the files in that library commit and push should be done automatically. is it something like it right.

Something like.

Also read on the existing git feature at https://blog.jabref.org/2025/11/11/introducing-git-integration-in-jabref/
